src/main/resource-applier.ts

```diff
@@ -110,47 +105,50 @@ export class ResourceApplier {
     const kubectl = await this.cluster.ensureKubectl();
     const kubectlPath = await kubectl.getPath();
     const proxyKubeconfigPath = await this.cluster.getProxyKubeconfigPath();
+    const tmpDir = tempy.directory();
 
-    return new Promise((resolve, reject) => {
-      const tmpDir = tempy.directory();
-
-      // Dump each resource into tmpDir
-      resources.forEach((resource, index) => {
-        fs.writeFileSync(path.join(tmpDir, `${index}.yaml`), resource);
-      });
-      args.push("-f", `"${tmpDir}"`);
-      const cmd = `"${kubectlPath}" ${subCmd} --kubeconfig "${proxyKubeconfigPath}" ${args.join(" ")}`;
-
-      logger.info(`[RESOURCE-APPLIER] running cmd ${cmd}`);
-      exec(cmd, (error, stdout) => {
-        if (error) {
-          logger.error(`[RESOURCE-APPLIER] cmd errored: ${error}`);
-          const splitError = error.toString().split(`.yaml": `);
-
-          if (splitError[1]) {
-            reject(splitError[1]);
-          } else {
-            reject(error);
-          }
-
-          return;
-        }
-
-        resolve(stdout);
-      });
-    });
-  }
+    const results = await Promise.allSettled(
+      resources.map((resource, index) => fse.writeFile(path.join(tmpDir, `${index}.yaml`), resource)),
+    );
+
+    const firstError = results.find(result => result.status === "rejected") as PromiseRejectedResult;
 
-  protected sanitizeObject(resource: KubernetesObject | any) {
-    resource = cloneJsonObject(resource);
-    delete resource.status;
-    delete resource.metadata?.resourceVersion;
-    const annotations = resource.metadata?.annotations;
+    if (firstError) {
+      await Promise.allSettled(
```

Comment:
You could just delete `tmpDir`? Though I would've thought that it was cleaned up automatically via`tempy`. Except looking at the docs it's not clear. If you use `tempy.directory.task` for sure the temp dir is cleaned.

My initial concern about the `promise.all()` was that if fse.writeFile() threw an error you'd just have a rejected promise in the results, which weren't checked. Execution would've continued to do the kubectl apply, no?

Reply:
`promise.all()` rejects as soon as any of the promises rejects and only resolves if all of them do

Reply:
FYI I have now moved it so that we always clean up the folder, even on success.